helm/kubernetes-cluster-prep/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,14 @@
+{{- if .Values.authnK8s.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.authnK8s.serviceAccount.name }}
```

Comment:
This needs to be:
```
 name: {{ .Values.authnK8s.serviceAccount.name | default "authn-k8s-serviceaccount" }}
```
This is another case where we want a "conditional default". In this case, we only want to provide a default when `Values.authnK8s.serviceAccount.create` is set to `true`. Since this default definition is made conditional by the `if .Values.authnK8s.serviceaccount.create` on Line 1.

Conversely, when the user "brings their own" ServiceAccount (`.Values.authnK8s.service.create` is `false`), then we DEFINITELY don't want to default the ServiceAccount name, because we need the user to tell us what their ServiceAccount name is that they want us to use. If we put defaults into the `values.yaml` for this setting, it would be an unconditional default.